### PR TITLE
Laser and Blast drills accept cryofluid as coolant

### PR DESCRIPTION
**Pull request by mnrsU**
Updated version of https://github.com/Anuken/Mindustry/pull/2219

based on https://github.com/Anuken/Mindustry-Suggestions/issues/277

Now only laser and blast drills accept cryo-cooling, mechanicals and pneumatics would just freeze. (change based on comment from xXextremeduro69420Xx)

### CODE REVIEW

**Comment by stuffyAI** [comment]
mechanical drills shouldn't be able to accept liquids?

**Comment by gen-NA-owl** [comment]
> mechanical drills shouldn't be able to accept liquids?

No they can accept water but not cryofluid

**Comment by TheSliverNickel** [comment]
What happened here?

**Comment by gen-NA-owl** [comment]
> Updated version of #2219
> 
> based on [Anuken/Mindustry-Suggestions#277](https://github.com/Anuken/Mindustry-Suggestions/issues/277)
> 
> Now only laser and blast drills accept cryo-cooling, mechanicals and pneumatics would just freeze (but they still can accept water) . (change based on comment from xXextremeduro69420Xx)

**Comment by ILiekMelons** [comment]
👀

**Comment by mnrsU** [comment]
hey It works

File: core/src/mindustry/world/blocks/production/Drill.java [hunk 1]
```diff
@@ -287,6 +294,15 @@ public void updateTile(){
 
                 drillEffect.at(getX() + Mathf.range(size), getY() + Mathf.range(size), dominantItem.color);
             }
+
+
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/production/Drill.java:298]: unnecessary empty line

File: core/src/mindustry/world/meta/values/DrillBoosterListValue.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    float drillSpeedRate = (baseDrillSpeed ? 1f : 0f) + maxUsed * multiplier * liquid.heatCapacity;
+                    float standardDrillSpeed = baseDrillSpeed ? drillSpeed : drillSpeed / (maxUsed * multiplier * 0.4f);
+                    float result = standardDrillSpeed / (drillSpeed / drillSpeedRate);
+                    bt.add(Core.bundle.format("blocks.drillboostspeed", Strings.fixed(result, 1)));
+                }).left().padTop(-9);
+                c.row();
+            }
+        }).colspan(table.getColumns());
+        table.row();
+
```
Comment (Quezler) [inline core/src/mindustry/world/meta/values/DrillBoosterListValue.java:49]: here to

File: core/src/mindustry/world/blocks/production/Drill.java [hunk 3]
```diff
@@ -255,12 +260,14 @@ public void updateTile(){
 
             timeDrilled += warmup * delta();
 
+
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/production/Drill.java:263]: and here

File: core/src/mindustry/world/blocks/production/Drill.java [hunk 4]
```diff
@@ -16,7 +16,11 @@
 import mindustry.type.*;
 import mindustry.ui.*;
 import mindustry.world.*;
+import mindustry.world.consumers.*;
+import mindustry.world.blocks.*;
 import mindustry.world.meta.*;
+import mindustry.world.meta.values.*;
+
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/production/Drill.java:23]: and also here

**Review by Quezler — CHANGES_REQUESTED** [review]

File: core/src/mindustry/world/blocks/production/Drill.java (hunk 1 above)
Comment (mnrsU) [reply core/src/mindustry/world/blocks/production/Drill.java]: All right, I made a commit to remove extra whitespace

**Comment by Syncinus** [comment]
This would be really useful in a lot of cases, and it makes a bunch of sense in general, as everything else accepts cryo and water. IMO though cryo should still be accepted by mechanical and pneumatic for consistency.

**Comment by mnrsU** [comment]
https://github.com/Anuken/Mindustry/pull/2861

**Pull request closed without merging**